composables/api/useGetCategoriesForLeaderboard/index.ts

```diff
@@ -0,0 +1,34 @@
+import { useRuntimeConfig } from '#app'
+import { ref } from 'vue'
+import useApi, { type optionalParameters } from '~/composables/useApi'
+import { Categories } from '~/lib/api/Categories'
+import type {
+  CategoryViewModelListView,
+  GetCategoriesForLeaderboardParams,
+} from '~/lib/api/data-contracts'
+
+export default async function useGetCategoriesForLeaderboard(
+  params: GetCategoriesForLeaderboardParams,
+  opts: optionalParameters<CategoryViewModelListView> = {},
+) {
+  const { onError, onOkay } = opts
+  const responseData = ref<CategoryViewModelListView>({
+    data: [],
+    limitDefault: 0,
+    limitMax: 0,
+    total: 0,
+  })
```

Comment:
I don't know if I like this pattern of having dummy default data here. Is it necessary? (See other api composables too.)

Reply:
Thinking this through; I agree that we shouldn't have dummy default data for our composables. I'll get rid of the ones related to this PR, and then we could start an issue for all other composables.